addon/components/hyper-table.js

```diff
@@ -45,13 +47,16 @@ export default Component.extend({
   _orderedFilteredColumns: computed(
     '_columns',
     '_availableColumnsKeyword',
+    '_activeColumnCategory',
     function() {
       let columns = A(this._columns);
 
-      if (!isEmpty(this._availableColumnsKeyword)) {
-        let reg = RegExp(this._availableColumnsKeyword, 'i');
-        columns = A(columns.filter((x) => reg.test(x.title)));
-      }
+      columns = A(columns.filter((x) => {
+        const hasKeyword = x.title.toLowerCase().indexOf(this._availableColumnsKeyword.toLowerCase()) >= 0;
+        const hasActiveGroup = x.categories.indexOf(this._activeColumnCategory) >= 0 || !this._activeColumnCategory;
```

Comment:
maybe the inverse `!this._activeColumnCategory || x.categories.indexOf(this._activeColumnCategory) >= 0` the check if there is an activeColumn is less expensive